### PR TITLE
Select active env even if not in list of all envs

### LINKED ISSUE #9718: Active Interpreter is no longer selected as the preferred kernel
Upstream issue https://github.com/microsoft/vscode-python/issues/18955

### PR DESCRIPTION
**Pull request by DonJayamanne**
Fixes #9718

### CODE REVIEW

File: src/kernels/helpers.ts [hunk 1]
```diff
@@ -230,7 +244,7 @@ export function findPreferredKernel(
         // Remember, all we're doing is sorting the list, just because its sorted in order of preference doesn't mean we have a match.
         preferredKernel = undefined;
     } else {
-        traceInfoIfCI(`Preferred kernel is ${JSON.stringify(kernels[0])}`);
+        traceInfoIfCI(`Preferred kernel is ${JSON.stringify(preferredKernel)}`);
```
Comment (DonJayamanne) [inline src/kernels/helpers.ts:247]: Was logging the wrong thing.

**Comment by rchiodo** [comment]
Wow, seems like a bug on python? How could the active interpreter not be an interpreter?

**Review by rchiodo — APPROVED** [review]

**Comment by DonJayamanne** [comment]
> How could the active interpreter not be an interpreter?

@rchiodo I think this is the same bug as https://github.com/microsoft/vscode-jupyter/issues/9703
Not entirely sure, once we merge this change & get a new insider release we can find out (tomorrow)

File: src/test/datascience/kernel-launcher/localKernelFinder.unit.test.ts [hunk 2]
```diff
@@ -1129,15 +1134,24 @@ import { JupyterServerUriStorage } from '../../../kernels/jupyter/launcher/serve
...
+                                },
+                                {
+                                    interpreter: python38VenvEnv,
+                                    kernelSpecs: [fullyQualifiedPythonKernelSpec]
                                 }
-                                // },
-                                // {
-                                // interpreter: python38VenvEnv,
-                                // kernelSpecs: [fullyQualifiedPythonKernelSpec]
-                                // }
```
Comment (DonJayamanne) [inline src/test/datascience/kernel-launcher/localKernelFinder.unit.test.ts:1137]: Not sure why this was disabled, enabling this.
The more intepreters we have in our test, the better. yes it complicates the tests.
But having more interpreters is a valid scenario.

**Pull request merged**